### PR TITLE
[Core] Rename NVIDIA_TESLA_A100 to NVIDIA_A100

### LINKED ISSUE #40767: [core] A100 Accelerator is labeled incorrectly
In terms of terminology confusion take a look at <https://github.com/ray-project/ray/blob/master/python/ray/util/accelerators/accelerators.py|this file> - contents reproduced below:
```NVIDIA_TESLA_V100 = "V100"
NVIDIA_TESLA_P100 = "P100"
NVIDIA_TESLA_T4 = "T4"
NVIDIA_TESLA_P4 = "P4"
NVIDIA_TESLA_K80 = "K80"
NVIDIA_TESLA_A100 = "A100"
NVIDIA_TESLA_A10G = "A10G"
INTEL_MAX_1550 = "Intel-GPU-Max-1550"
INTEL_MAX_1100 = "Intel-GPU-Max-1100"
AWS_NEURON_CORE = "aws-neuron-core"
GOOGLE_TPU_V2 = "TPU-V2"
GOOGLE_TPU_V3 = "TPU-V3"
GOOGLE_TPU_V4 = "TPU-V4"```

A100 is not part of the "TESLA" suite -- probably better to label it as NVIDIA_A100 instead of NVIDIA_TESLA_A100

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
A100 is not part of Tesla suite.

Still keep the old variable name for backward compatibility.

```
>>> accelerators.NVIDIA_TESLA_A100
<stdin>:1: RayDeprecationWarning: NVIDIA_TESLA_A100 is deprecated, use NVIDIA_A100 instead.
'A100'
>>> accelerators.NVIDIA_A100
'A100'
>>> accelerators.NVIDIA_A100f
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jjyao/Workspace/ray2/python/ray/util/accelerators/__init__.py", line 48, in __getattr__
    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
AttributeError: module 'ray.util.accelerators' has no attribute 'NVIDIA_A100f'
```

## Related issue number
Closes #40767

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
```python3
class WarnOnAccess:
    def __init__(self, variable, warning_message="Warning: Variable accessed."):
        self.variable = variable
        self.warning_message = warning_message

    def __getattr__(self, attr):
        # Print a warning message when the variable is accessed
        print(self.warning_message)
        return getattr(self.variable, attr)

# Example usage
my_variable = WarnOnAccess(42, "Accessing a variable.")
print(my_variable)
print(my_variable + 10)
```

Q: should we do something like this to make warning more clear? the code is from chatgpt

File: python/ray/util/accelerators/__init__.py [hunk 1]
```diff
@@ -20,12 +22,27 @@
...
+    if name == "NVIDIA_TESLA_A100":
+        from ray.util.annotations import RayDeprecationWarning
+
+        warnings.warn(
+            "NVIDIA_TESLA_A100 is deprecated, use NVIDIA_A100 instead.",
+            RayDeprecationWarning,
+            stacklevel=2,
+        )
+        return NVIDIA_A100
+    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```
Comment (rkooo567) [inline python/ray/util/accelerators/__init__.py:48]: hmm how  does it work when other name is given in this case? (iiuc it should raise an exception?)

File: python/ray/tests/test_accelerators.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import os
```
Comment (rkooo567) [inline python/ray/tests/test_accelerators.py:1]: put the test under tests/acclerator folder?

**Review by rkooo567 — APPROVED** [review]
lgtm

File: python/ray/util/accelerators/__init__.py (hunk 1 above)
Comment (jjyao) [reply python/ray/util/accelerators/__init__.py:48]: It won't

**Pull request merged**